c_src/sasl_auth.c

```diff
@@ -271,17 +466,26 @@ static ERL_NIF_TERM sasl_kinit(ErlNifEnv* env, int UNUSED(argc), const ERL_NIF_T
             || !enif_inspect_binary(env, argv[1], &principal_in)))
         return enif_make_badarg(env);
 
+    principal_mut = copy_bin(principal_in);
```

Comment:
Similar to not utilizing transient arguments with cyrus-sasl functions, we simply do the same here, although this may seem like overkill, we definitely do not want to the functions to have a chance to mutate what we pass in. In a nutshell, it's a safety measure.